core/drivers/stm32_uart.c

```diff
@@ -149,7 +152,19 @@ struct stm32_uart_pdata *stm32_uart_init_from_dt_node(void *fdt, int node)
 	pd->chip.ops = &stm32_uart_serial_ops;
 	pd->base.pa = info.reg;
 	pd->secure = (info.status == DT_STATUS_OK_SEC);
-	pd->clock = (unsigned int)info.clock;
+
+	if (fdt_getprop(fdt, node, "clocks", NULL)) {
```

Comment:
Clocks is part of the mandatory properties listed in the kernel Documentation/devicetree/bindings/serial/st,stm32-uart.yaml.
We do not expect the uart to not have clocks property here.

Reply:
Ok, i'll remove the test.